### PR TITLE
Reset password flow

### PR DESCRIPTION
**Pull request by marcoacierno**
image

### CODE REVIEW

File: users-backend/users/api/mutations/request_reset_password.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    user = await info.context.users_repository.get_by_email(email)
+
+    if not user:
+        logger.info("Trying to request reset password of not existent user")
+        return OperationSuccess(ok=True)
+
+    try:
+        await services.request_reset_password(user)
+    except UserIsNotActiveError:
+        pass
```
Comment (patrick91) [inline users-backend/users/api/mutations/request_reset_password.py:28]: why pass here? should we add some logs in any case?

File: users-backend/users/domain/entities.py [hunk 2]
```diff
@@ -91,6 +112,13 @@ def display_name(self) -> str:
     Column("is_active", Boolean(), default=True, nullable=False),
     Column("is_staff", Boolean(), default=False, nullable=False),
     Column("is_superuser", Boolean(), default=False, nullable=False),
+    Column(
+        "request_reset_password_id",
```
Comment (patrick91) [inline users-backend/users/domain/entities.py:116]: I'd make this a bit more generic as we can use also to invalidate existing login tokens (when someone logs out from everywhere, when they change password, when we ban them)

Comment (marcoacierno) [reply users-backend/users/domain/entities.py:116]: sure

Comment (marcoacierno) [reply users-backend/users/domain/entities.py:116]: consider that every time the user sends a request to our gateway we get information about the user, so it's easier for us invalidating tokens and so on

File: users-backend/users/api/mutations/request_reset_password.py (hunk 1 above)
Comment (marcoacierno) [reply users-backend/users/api/mutations/request_reset_password.py:28]: we already log it in the service

File: users-backend/users/domain/entities.py [hunk 3]
```diff
@@ -63,6 +65,24 @@ def set_password(self, raw_password: str):
     def has_usable_password(self) -> bool:
         return is_password_usable(self.hashed_password)
 
+    def get_reset_password_jwt_id(self) -> str:
```
Comment (marcoacierno) [inline users-backend/users/domain/entities.py:68]: test this

File: users-backend/users/domain/entities.py [hunk 4]
```diff
@@ -63,6 +65,24 @@ def set_password(self, raw_password: str):
...
+        return f"reset-password:{self.id}:{self.jwt_auth_id}"
+
+    def create_reset_password_token(self) -> str:
```
Comment (marcoacierno) [inline users-backend/users/domain/entities.py:71]: test this

File: users-backend/users/domain/services/__init__.py [hunk 5]
```diff
@@ -13,4 +15,9 @@
     "social_login",
     "SocialLoginInput",
     "SocialAccount",
+    # reset_password
+    "reset_password",
+    "ResetPasswordInput",
+    # request_reset_password
```
Comment (estyxx) [inline users-backend/users/domain/services/__init__.py:21]: this comments now can be removed....

**Pull request merged**